File: 프로그래머스/lv1/131705. 삼총사/삼총사.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> number)
{
    int answer = 0;
    for (int i = 0; i < number.size() - 1; i++)
    {
        for (int j = i + 1; j < number.size(); j++)
        {
            int temp = number[i] + number[j];
            vector<int> temp2;
            temp2 = number;
            temp2.erase(temp2.begin() + i);
            temp2.erase(temp2.begin() + j - 1);
            for (int z = 0; z < temp2.size(); z++)
            {
                if (temp + temp2[z] == 0)
                    answer++;
            }
            
        }
    }

    return answer / 3;
}

File: 프로그래머스/1/131705. 삼총사/삼총사.java
import java.util.*;

class Solution {
    
    static int answer = 0;
    public int solution(int[] number) {
        go(number, new boolean[number.length], 0, 0, 0);
        return answer;
    }
    
    private void go(int[] number, boolean[] visited, int r, int depth, int sum) {
        if (depth == 3) {
            if (sum == 0) answer++; 
            return;
        }
        
        for (int i = r; i < number.length; i++) {
            go(number, visited, i + 1, depth + 1, sum + number[i]);
        }
    }
}